.github/workflows/distro.yml

```diff
@@ -0,0 +1,323 @@
+
+
+# https://stackoverflow.com/a/73467112
+    - name: Split os
+      shell: bash
+      env:
+        OS: ${{ matrix.OS }}
+      id: split
+      run: echo "platform_version=${OS##*-}" | tee -a $GITHUB_OUTPUT
+
+    - name: Install boost
+      uses: MarkusJx/install-boost@v2.4.4
+      id: install-boost
+      with:
+        boost_version: 1.72.0
+        platform_version: ${{ steps.split.outputs.platform_version }}
+        arch: ${{ matrix.ARCH == 'x86_64' && 'x86' || 'aarch64' }}
+
+    - name: cibuildwheel
+      if: ${{ matrix.OS != 'ubuntu-20.04' || matrix.ARCH != 'aarch64' }}
+      shell: bash
+      run: |
+        
+        pip install cibuildwheel
+        
+        PIP_FIND_LINKS="https://makslevental.github.io/wheels https://github.com/Xilinx/mlir-aie/releases/expanded_assets/latest-wheels" \
+        BOOST_ROOT="${{ steps.install-boost.outputs.BOOST_ROOT }}" \
+        CIBW_ARCHS=${{ matrix.ARCH }} \
+        CMAKE_GENERATOR="Ninja" \
+        HOST_CCACHE_DIR=${{ steps.configure_ccache_dir_on_host.outputs.HOST_CCACHE_DIR }} \
+        MATRIX_OS=${{ matrix.OS }} \
+        MLIR_COMMIT="17.0.0.2023083019+35ca6498" \
+        PARALLEL_LEVEL=${{ matrix.OS == 'windows-2019' && '2' || '4' }} \
+        cibuildwheel --output-dir wheelhouse
+
+    - name: build aarch ubuntu wheel
+      shell: bash
+      if: ${{ matrix.OS == 'ubuntu-20.04' && matrix.ARCH == 'aarch64' }}
+      run: |
+        
+        export PIP_NO_BUILD_ISOLATION="false"
+        export PIP_FIND_LINKS="https://makslevental.github.io/wheels https://github.com/Xilinx/mlir-aie/releases/expanded_assets/latest-wheels"
+        export CIBW_ARCHS=${{ matrix.ARCH }}
+        export MLIR_COMMIT="17.0.0.2023083019+35ca6498"
+        export BOOST_ROOT="${{ steps.install-boost.outputs.BOOST_ROOT }}"
+        export MATRIX_OS=${{ matrix.OS }}
+        
+        pip install -r requirements.txt
+        ./scripts/pip_install_mlir.sh
+        
+        CMAKE_GENERATOR=Ninja \
+        pip wheel . -v -w wheelhouse --no-build-isolation
+
+    - name: Get wheel version
+      id: get_wheel_version
+      working-directory: ${{ steps.setup_base.outputs.WORKSPACE_ROOT }}
+      shell: bash
+      run: |
+        
+        pip install pkginfo
+        WHL=$(ls wheelhouse/mlir_aie-*whl)
+        echo "MLIR_AIE_WHEEL_VERSION=$(python -c "import pkginfo; print(pkginfo.Wheel('$WHL').version)")" | tee -a $GITHUB_OUTPUT
+
+    - name: rename
+      shell: bash
+      if: ${{ matrix.OS == 'ubuntu-20.04' || matrix.OS == 'macos-11' }}
+      run: |
+        if [ x"${{ matrix.OS }}" == x"macos-11" ]; then
+          brew install rename
+        else
+          sudo apt-get install -y rename
+        fi
+        rename 's/cp311-cp311/py3-none/' wheelhouse/mlir_aie-*whl
+        
+        if [ x"${{ matrix.OS }}" == x"ubuntu-20.04" ] && [ x"${{ matrix.ARCH }}" == x"aarch64" ]; then
+          rename 's/x86_64/aarch64/' wheelhouse/mlir_aie-*whl
+        fi
+
+    - name: rename
```

Comment:
Name conflict with previous step.

Reply:
They never execute on the same runner...

Reply:
Whatever, this is confusing in https://github.com/Xilinx/mlir-aie/actions/runs/6103552291/job/16564151441?pr=626 with 2 "rename".